daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/IdeLedgerClient.scala

```diff
@@ -226,9 +227,18 @@ class IdeLedgerClient(
       ec: ExecutionContext,
       mat: Materializer,
   ): Future[Option[ScriptLedgerClient.ActiveContract]] = {
+    val keyValue = key.toUnnormalizedValue
+    // TODO: Can this be any smarter? Are we able to update the exception thrown by Hash? Should we instead check this ourselves
+    // Alternatively, can we assume that the only error `GlobalKey.build` might give is the contractId exception?
+    def keyBuilderError(err: String): Future[GlobalKey] =
+      Future.failed(
+        if (err == "Contract IDs are not supported in contract keys.")
+          SError.SErrorDamlException(ContractIdInContractKey(keyValue))
+        else new ConverterException(err)
+      )
```

Comment:
> can we assume that the only error `GlobalKey.build` might give is the contractId exception

This seems true from looking at the source of `GlobalKey`, but ignoring the string seems iffy since we might introduce other failure modes later. I think we'd want a version of `GlobalKey.build` that either throws custom exceptions or returns its own error type in a `Left`, so we can catch/match on those, but it can be done later.

Reply:
Not 100% sure, but I think it is better to assume it can throws. (Maybe we need a test).

Reply:
I'll look into cleaning this up a little - I think a custom error sum on the Left is best